### PR TITLE
Rewrite asset file paths to use stable meaningful names

### LINKED ISSUE #198: For Asset filenames, use name ("MyImage.svg") instead of number (0003.svg)
Similar to what we did with control filenames to use control name instead of oid. 
(this can be done in a non-breaking way)

### PR DESCRIPTION
**Pull request by lesaltzm**
fixes #198

### CODE REVIEW

**Comment by lesaltzm** [comment]
Bump the source version

File: src/PAModel/Utility/Utilities.cs [hunk 1]
```diff
@@ -328,5 +329,25 @@ public static void EnsurePathRooted(string path)
                 throw new ArgumentException("Path must be a full path: " + path);
             }
         }
+
+        /// <summary>
+        /// Relative path of the file depends on the fileType. eg: Image files go into "Assets\Images\"
```
Comment (MikeStall) [inline src/PAModel/Utility/Utilities.cs:334]: Can we infer this by file extension instead? (different directories is an unusual pattern for pro-devs)

File: src/PAModel/CanvasDocument.cs [hunk 2]
```diff
@@ -422,8 +426,85 @@ internal HashSet<string> GetImportedComponents()
...
+
+
+        private void StabilizeAssetFilePaths()
+        {
+            _entropy.LocalResourceFileNames.Clear();
+
+            // Update AssetFile paths
+            foreach (var resource in _resourcesJson.Resources)
+            {
+                if (resource.ResourceKind != "LocalFile")
```
Comment (MikeStall) [inline src/PAModel/CanvasDocument.cs:438]: nit: can we put this string literal as a constant? als omakes it easier to comment it.

File: src/PAModel/CanvasDocument.cs [hunk 3]
```diff
@@ -422,8 +426,85 @@ internal HashSet<string> GetImportedComponents()
...
+                    continue;
+
+                var assetFilePath = FilePath.FromMsAppPath(resource.Path.Substring(AssetFilePathPrefix.Length));
```
Comment (MikeStall) [inline src/PAModel/CanvasDocument.cs:441]: nit: can we wrap this part in a  function? We repeat it 4x, the substring on file path is brittle. 
```
FilePath.FromMsAppPath(resource.Path.Substring(AssetFilePathPrefix.Length));
```

File: src/PAModel/Utility/Utilities.cs (hunk 1 above)
Comment (lesaltzm) [reply src/PAModel/Utility/Utilities.cs:334]: Yes, absolutely. We'll still need this function for the msapp side, but we could dump them flat in Assets/

File: src/PAModel/CanvasDocument.cs (hunk 2 above)
Comment (lesaltzm) [reply src/PAModel/CanvasDocument.cs:438]: yup

File: src/PAModel/Utility/Utilities.cs (hunk 1 above)
Comment (lesaltzm) [reply src/PAModel/Utility/Utilities.cs:334]: Discussed offline, keeping.

**Review by MikeStall — APPROVED** [review]

**Pull request merged**